plugins/api/include/irods/plugins/api/switch_user_types.h

```diff
@@ -1,15 +1,63 @@
 #ifndef IRODS_SWITCH_USER_TYPES_H
 #define IRODS_SWITCH_USER_TYPES_H
 
+/// The input data type used to invoke rc_switch_user.
+///
+/// \since 4.3.1
 typedef struct SwitchUserInput // NOLINT(modernize-use-using)
 {
-    // NOLINTBEGIN(modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+
+    /// The part of a fully-qualified iRODS username preceding the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::userName) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the zone
```

Comment:
Is this true? `tempZone_user` cannot exist in zone `tempZone`? Or did I read this wrong?

Reply:
Your confusion indicates this wording isn't clear.

It is suppose to express that passing a string such as `"alice#tempZone"` is not allowed. Instead, `alice` and `tempZone` must be passed via the `username` and `zone` member variables.

Reply:
Ohhh, I see. I think this is fine. Not sure why I read it that way...

I don't have any suggestions for how to improve it.

Reply:
An example for `rc_switch_user` would help here. Will add one.